File: ios/Utils/Mapper/TypedBannerAdSizeMapper.swift
import Foundation

class TypedBannerAdSizeMapper {
    private let bannerAdSizeMapper: BannerAdSizeMapper
    
    convenience init() {
        let bannerAdSizeMapper = BannerAdSizeMapper()
        self.init(bannerAdSizeMapper: bannerAdSizeMapper)
    }
    
    init(bannerAdSizeMapper: BannerAdSizeMapper) {
        self.bannerAdSizeMapper = bannerAdSizeMapper
    }
    
    func map(bannedAdSize: TypedBannerAdSize) -> [String: Any] {
        var result = bannerAdSizeMapper.map(
            bannerAdSize: bannedAdSize.bannerAdSize,
            initialWidth: bannedAdSize.initialWidth,
            initialHeight: bannedAdSize.initialHeight
        )
        result[Constants.Key.bannerType] = bannedAdSize.type
        return result
    }
}

File: android/src/main/java/com/yandexmobileads/util/mapper/TypedBannerAdSizeMapper.kt
package com.yandexmobileads.util.mapper

import android.content.Context
import com.yandexmobileads.banner.TypedBannerAdSize
import com.yandexmobileads.common.Constants

internal class TypedBannerAdSizeMapper(
    context: Context,
    private val bannerAdSizeMapper: BannerAdSizeMapper = BannerAdSizeMapper(context),
) {

    fun map(bannerAdSize: TypedBannerAdSize): Map<String, Any> {
        val result = bannerAdSizeMapper.map(bannerAdSize.bannerAdSize).toMutableMap()
        result[Constants.Key.BANNER_TYPE] = bannerAdSize.type
        return result
    }
}
